src/view/video/SessionList.tsx

```diff
@@ -178,45 +190,85 @@ class SessionList extends React.Component<StyledComponentProps> {
     }
   }
 
-  private renderDescription(description: string) {
+  private renderDescription(description: string, keywordList: string[]) {
     const maxLines = 3;
     const maxChars = 140;
 
     const lines = description.split(/\r\n|\r|\n/);
-    const displayLines: string[] = [];
-    let count = 0;
-    for (const line of lines) {
-      if (displayLines.length >= maxLines) {
-        displayLines.push("…");
-        break;
+    const lineWithBoldRanges: ILineWithBoldRange[] = lines.map(line => this.detectBoldRange(line, keywordList))
+
+    const displayLines: ILineWithBoldRange[] = [];
+    const firstKeywordLine = lineWithBoldRanges.findIndex(lwbr => lwbr.boldRanges.length > 0);
+    if (firstKeywordLine === -1) {
+      // the case that no keywords are found in description
+      let count = 0;
+      for (const line of lines) {
+        if (displayLines.length >= maxLines) {
+          displayLines.push({line: "…", boldRanges: []});
+          break;
+        }
+
+        const length = line.length;
+        if (count + length < maxChars) {
+          count += length;
+          displayLines.push({line, boldRanges: []});
+        } else {
+          displayLines.push({line: line.substr(0, maxChars - count) + "…", boldRanges: []});
+          break;
+        }
       }
+    } else {
+      // the case that keywords are found in description
+      let count = 0;
+      for (let lineIndex = firstKeywordLine; lineIndex < lineWithBoldRanges.length; lineIndex++) {
+        if (displayLines.length >= maxLines) {
+          displayLines.push({line: "…", boldRanges: []});
+          break;
+        }
 
-      const length = line.length;
-      if (count + length < maxChars) {
-        count += length;
-        displayLines.push(line);
-      } else {
-        displayLines.push(line.substr(0, maxChars - count) + "…");
-        break;
+        let lwbr = lineWithBoldRanges[lineIndex];
+        if (lineIndex === firstKeywordLine) {
+          if (lwbr.boldRanges[0].offset >= maxChars) {
```

Comment:
```suggestion
          if (lwbr.boldRanges[0].offset > maxChars / 2) {
```
On the next line, `maxChars / 2` is aim to show sentences before and after the keyword.
So it would be nice to compare with `maxChars / 2` here, too.